### PR TITLE
Setup Cypress

### LINKED ISSUE #55: Setup Cypress
So, we need some infra for e2e tests. Cypress is the best option for that. 
Also, we need to setup [mocks for graphql](https://betterprogramming.pub/intercepting-graphql-requests-with-cypress-99984d6d500c).

**NB:** tests must works inside docker container #25 because of dependency on pg.

### PR DESCRIPTION
**Pull request by peter-nikitin**
resolve #55

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+COPY --chown=node:node . .
+
+USER node
+
+RUN npm ci
+RUN npx prisma generate
+RUN npm run build
+
+EXPOSE 3000
+CMD [ "npm","run", "start" ]
```
Comment (awinogradov) [inline Dockerfile:16]: Spaces and empty lines

File: cypress/e2e/home_page.cy.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+describe('The Home Page', () => {
+    it('successfully loads', () => {
+        cy.visit('/'); // change URL to match your dev URL
+        cy.wait(3000);
```
Comment (awinogradov) [inline cypress/e2e/home_page.cy.ts:4]: It was for debug only, currently you can drop this line

File: src/components/UserInviteForm.tsx [hunk 3]
```diff
@@ -89,14 +89,18 @@ export const UserInviteForm: React.FC<UserInviteFormProps> = ({ card, onCreate }
             <FormActions flat="top">
                 <FormAction left />
                 <FormAction right inline>
-                    <Button size="l" view="primary" type="submit" disabled={!isValid} text={t('Send invite')} />
+                    {/*
+ // @ts-ignore */}
```
Comment (awinogradov) [inline src/components/UserInviteForm.tsx:93]: Add #25 to every of ts-ignore comments plz. It must be fixed before production.

**Review by awinogradov — CHANGES_REQUESTED** [review]
It would be nice to add some instructions about local usage. Also, I think we need change `npm run db:run` command in `package.json` scripts

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,74 @@
+# Naively Simple Node Dockerfile
```
[the remark this answers is not in the document]
Comment (peter-nikitin) [reply Dockerfile]: done

File: Dockerfile (hunk 1 above)
Comment (peter-nikitin) [reply Dockerfile]: update Dockerfile

File: src/components/UserInviteForm.tsx (hunk 3 above)
Comment (peter-nikitin) [reply src/components/UserInviteForm.tsx:94]: done

File: cypress/e2e/home_page.cy.ts (hunk 2 above)
Comment (peter-nikitin) [reply cypress/e2e/home_page.cy.ts]: done

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,45 @@
+
+# START DEPS IMAGE
+
+FROM node:16-alpine as deps
```
Comment (awinogradov) [inline Dockerfile:4]: `AS` or `as`? ;)

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+# END BUILD IMAGE
+
+# START RUNNER IMAGE
+
+FROM node:16-alpine AS runner
+
+ENV NODE_ENV production
+RUN addgroup -g 1001 -S nodejs
+RUN adduser -S nextjs -u 1001
+RUN apk update && apk add  curl
```
Comment (awinogradov) [inline Dockerfile:29]: double space

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+WORKDIR /app
+COPY --from=build --chown=nextjs:nodejs /app/package.json /app/package-lock.json ./
+COPY --from=build --chown=nextjs:nodejs /app/node_modules ./node_modules
+COPY --from=build --chown=nextjs:nodejs /app/public ./public
+COPY --from=build --chown=nextjs:nodejs /app/.next ./.next
+COPY --from=build --chown=nextjs:nodejs /app/next.config.js  ./
+COPY --from=build --chown=nextjs:nodejs /app/prisma  ./prisma
```
Comment (awinogradov) [inline Dockerfile:37]: double space

File: docker-compose.yml [hunk 8]
```diff
@@ -17,3 +17,38 @@ services:
...
+    ports:
+      - 3000:3000
+    build:
+      context: .
+      dockerfile: Dockerfile
+    depends_on:
+      - db
+    environment:
+      - DATABASE_URL=postgresql://prisma:hilly-sand-pit@db:5432/prisma?schema=prisma-pg-test
+    # command: sh -c 'npx prisma migrate dev --preview-feature'
```
Comment (awinogradov) [inline docker-compose.yml:31]: this command moved to shell script

File: package.json [hunk 9]
```diff
@@ -3,15 +3,18 @@
   "version": "0.1.0",
```
Comment (awinogradov) [inline package.json:3]: Fucking prekrasno!

File: src/components/UserInviteForm.tsx [hunk 10]
```diff
@@ -89,14 +89,19 @@ export const UserInviteForm: React.FC<UserInviteFormProps> = ({ card, onCreate }
...
+                    <Button size="l" view="primary-outline" type="submit" disabled={!isValid} text={t('Send invite')} />
                 </FormAction>
             </FormActions>
         </Form>
     );
 
     return (
         <>
+            {/*
```
Comment (awinogradov) [inline src/components/UserInviteForm.tsx:103]: smthg strange)

File: src/pages/users/invite.tsx [hunk 11]
```diff
@@ -8,6 +8,8 @@ function Page() {
     const t = useTranslations('users.invite');
 
     return (
+        // FIXME: https://github.com/taskany-inc/issues/issues/25
+        // @ts-ignore
```
Comment (awinogradov) [inline src/pages/users/invite.tsx:12]: You can do it in one line 

``` ts
// @ts-ignore FIXME: https://github.com/taskany-inc/issues/issues/25
```

It's ok. Minor thing.

File: Dockerfile (hunk 7 above)
Comment (peter-nikitin) [reply Dockerfile:37]: fix

File: src/components/UserInviteForm.tsx (hunk 10 above)
Comment (peter-nikitin) [reply src/components/UserInviteForm.tsx:103]: remove

File: docker-compose.yml (hunk 8 above)
Comment (peter-nikitin) [reply docker-compose.yml:31]: fix

File: Dockerfile (hunk 6 above)
Comment (peter-nikitin) [reply Dockerfile:29]: fix

File: package.json (hunk 9 above)
Comment (peter-nikitin) [reply package.json:3]: ❤️

**Review by awinogradov — APPROVED** [review]
🔥

**Pull request merged**